extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RemoveOptionalFromConfigMapEnvSourceDecorator.java

```diff
@@ -14,6 +18,8 @@ public void andThenVisit(ConfigMapEnvSourceFluent ref) {
 
     @Override
     public Class<? extends Decorator>[] after() {
-        return new Class[] { AddEnvVarDecorator.class, ApplicationContainerDecorator.class };
+        return new Class[] { AddEnvVarDecorator.class,
+                AddSecretVolumeDecorator.class, AddSecretVolumeToRevisionDecorator.class,
+                AddConfigMapVolumeToRevisionDecorator.class, AddConfigMapVolumeDecorator.class };
```

Comment:
To fix the cycle issue, I realized that there were no ApplicationContainerDecorator types that adds this optional flag. However, I added the decorators that do set this value instead.